quarkchain/cluster/shard_state.py

```diff
@@ -671,7 +690,8 @@ def add_block(self, block, skip_if_too_old=True):
                     )
                 )
 
-        if self.db.contain_minor_block_by_hash(block.header.get_hash()):
+        header_hash = block.header.get_hash()
```

Comment:
I don't prefer pre-mature optimization, especially such variable is used multiple places in the method and it is hard to find its context when reading its references.